### PR TITLE
Add fixes to allow carbon simulations to run with a single advected species

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/reference/CONTRIBUTING.html)

### Describe the update

This PR adds several updates & fixes to allow the carbon simulations to use only a single advected species.  A new script `run/shared/singleCarbonSpecies` is called to edit the "out-of-the-box" run directory configuration files (which by default advect CH4, CO, CO2, and OCS) so that only the species of interest is advected (and that emissions/diagnostics for the other species are disabled).  

The `singleCarbonSpecies.sh` script is called from the `createRunDir.sh` script for both GCClassic and GCHP, which now asks if you wish to use all species or a single species:

```console
===========================================================
GEOS-CHEM RUN DIRECTORY CREATION
===========================================================

-----------------------------------------------------------
Choose simulation type:
-----------------------------------------------------------
   1. Full chemistry
   2. Aerosols only
   3. CH4
   4. CO2
   5. Hg
   6. POPs
   7. Tagged CH4
   8. Tagged CO
   9. Tagged O3
  10. TransportTracers
  11. Trace metals
  12. Carbon
>>> 12

-----------------------------------------------------------
Do you wish to use a single advected species?
-----------------------------------------------------------
  1. Use all species
  2. Use CH4 only
  3. Use CO2 only
  4. Use CO only
  5. Use OCS only
>>> 
```
Also, an integration test has been added (for GEOS-Chem Classic only at this time, GCHP to follow) for the carbon simulation with CH4 as the only advected species.

### Expected changes

This is a zero-diff update w/r/t the fullchem benchmark simulations.  It will only affect the carbon simulation by giving the user the option to advect a single species.

### Reference(s)

N/A

### Related Github Issue(s)

N/A

### CODE REVIEW

**Comment by yantosca** [comment]
GEOS-Chem Classic integration tests on the `bugfix/single-carbon-tracer` branch all passed, except for TOMAS, which is a known issue that will be fixed in 14.3.0.
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #f73eb7e GEOS-Chem submod update: Merge PR #1957 and PR #1958
GEOS-Chem #24a6b77ef Update int test scripts for single-species carbon simulation
HEMCO     #daf54d0 Merge PR #235 (Bug fix for vertical regridding)

Using 24 OpenMP threads
Number of execution tests: 27

Submitted as SLURM job: 2217582
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
...
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....FAIL
gc_4x5_47L_merra2_fullchem_TOMAS40..................Execute Simulation....FAIL
...
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 25
Execution tests failed: 2
Execution tests not yet completed: 0
```
All GCHP integration tests also passed:
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #d5180cc GEOS-Chem submod update: Merge PR #1957 and PR #1958
GEOS-Chem #24a6b77ef Update int test scripts for single-species carbon simulation
HEMCO     #daf54d0 Merge PR #235 (Bug fix for vertical regridding)

Number of execution tests: 5

Submitted as SLURM job: 2217719
==============================================================================

...
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 5
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

```

**Comment by yantosca** [comment]
Also tagging @hannahnesser @kbowman77 @dbajones for your reference.

**Comment by yantosca** [comment]
Integration tests on `bugfix/carbon-single-tracer` locally merged into `dev/14.2.1` are now running.

**Comment by yantosca** [comment]
After merging locally atop PR #1957 and PR #1958, all GEOS-Chem Classic integration tests passed, except TOMAS (which will be fixed later).
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #f73eb7e GEOS-Chem submod update: Merge PR #1957 and PR #1958
GEOS-Chem #155c6b66d Merge PR #1963 (Fixes to allow carbon sims w/ a single advected species)
HEMCO     #daf54d0 Merge PR #235 (Bug fix for vertical regridding)

Using 24 OpenMP threads
Number of execution tests: 27

Submitted as SLURM job: 2284433
==============================================================================
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 25
Execution tests failed: 2
Execution tests not yet completed: 0
```
Also all integration tests were zero-diff w/r/t the prior integration test, except for these:
```console
Checking gc_4x5_merra2_carbon
   -> 1 difference found in OutputDir
      * GCC_r24/rundirs/gc_4x5_merra2_carbon/OutputDir/HEMCO_diagnostics.201901010000.nc 
        GCC_r25/rundirs/gc_4x5_merra2_carbon/OutputDir/HEMCO_diagnostics.201901010000.nc 

Checking gc_4x5_merra2_fullchem_APM
   -> 2 differences found in OutputDir
      * GCC_r24/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.Metrics.20190701_0000z.nc4 
        GCC_r25/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.Metrics.20190701_0000z.nc4 
      * GCC_r24/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.SpeciesConc.20190701_0000z.nc4 
        GCC_r25/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.SpeciesConc.20190701_0000z.nc4 
   -> 1 difference found in Restarts
      * GCC_r24/rundirs/gc_4x5_merra2_fullchem_APM/Restarts/GEOSChem.Restart.20190701_0100z.nc4 
        GCC_r25/rundirs/gc_4x5_merra2_fullchem_APM/Restarts/GEOSChem.Restart.20190701_0100z.nc4 

Checking gc_4x5_merra2_fullchem_RRTMG
   -> 1 difference found in OutputDir
      * GCC_r24/rundirs/gc_4x5_merra2_fullchem_RRTMG/OutputDir/GEOSChem.RRTMG.20190701_0000z.nc4 
        GCC_r25/rundirs/gc_4x5_merra2_fullchem_RRTMG/OutputDir/GEOSChem.RRTMG.20190701_0000z.nc4 
```
The difference in the carbon simulation is because in this PR, we have now activated the CO2 biomass burning for GFED2, where it was deactivated before:
```console
< 
< #NOTE: Biomass burning doesn't seem to work.  Keep here for now.
< # Uncomment this if GFED is used
< #EmisCO2_Biomass_GFED  CO2    111    1    1   2  kg/m2/s  CO2_biomass_burning_emissions_from_GFED4
< 
< # Uncomment this if FINN is used
---
> EmisCO2_Biomass_GFED  CO2    111    1    1   2  kg/m2/s  CO2_biomass_burning_emissions_from_GFED4
75,148d68
```

**Review by yantosca — COMMENTED** [review]
I will also push the following changes as a post-release commit:
```console
commit c8e0ebefaa143c6d36289858a3fd531c78d9b8ad (HEAD -> dev/14.2.1)
Author: Bob Yantosca <yantosca@seas.harvard.edu>
Date:   Fri Sep 15 12:39:00 2023 -0400

    PR #1963 post-merge fix: Change Emis*_Bioburn to Emis*_BiomassBurn
    
    run/GCClassic/HEMCO_Diagn.rc.templates/HEMCO_Diagn.rc.carbon
    run/GCHP/HEMCO_Diagn.rc.templates/HEMCO_Diagn.rc.carbon
    - Renamed EmisCO_Bioburn to EmisCO_BiomassBurn
    - Renamed EmisCO2_Bioburn to EmisCO2_BiomassBurn
    - Renamed EmisOCS_Bioburn to EmisOCS_BiomassBurn
    - Now use Cat = -1 and Hier = -1 for Emis*_BiomassBurn containers
      as well as extension 111 (GFED)
    
    run/GCHP/HISTORY.rc.templates/HISTORY.rc.carbon
    - Renamed EmisCO_Bioburn to EmisCO_BiomassBurn
    - Renamed EmisCO2_Bioburn to EmisCO2_BiomassBurn
    - Added OCS emission diagnostics
    
    Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -131,6 +134,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fixed entries for CH4 emissions in `HEMCO_Config.rc.carbon`
 
 ## [14.1.1] - 2023-03-03
+>>>>>>> 4722f288e90291ba904222f4bbe4fc216d17c34a
```
Comment (msulprizio) [inline CHANGELOG.md:137]: Remove this line

File: run/GCClassic/createRunDir.sh [hunk 2]
```diff
@@ -796,6 +824,7 @@ done
 
 mkdir -p ${rundir}
 mkdir -p ${rundir}/Restarts
+ls ${rundir}/Restarts
```
Comment (msulprizio) [inline run/GCClassic/createRunDir.sh:827]: Is this line meant to be here or is it leftover from debugging?

File: run/shared/singleCarbonSpecies.sh [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+#   # If CO is in the exclude list, turn off CO options
+#    isItemInList "CO2" "${1}"
+#    if [[ $? == 0 ]]; then
+#        keys=("use_fullchem_PCO_from_CH4"   \
+#              "use_fullchem_PCO_from_NMVOC")
+#        for key in ${keys[@]}; do
+#            keyValueUpdate "${key}" "true" "false" "${file}"
+#        done
+#    fi
```
Comment (msulprizio) [inline run/shared/singleCarbonSpecies.sh:124]: Can these lines be removed if they're commented out?

**Review by msulprizio — COMMENTED** [review]
I just had some minor questions and comments. See below.

File: run/GCClassic/createRunDir.sh (hunk 2 above)
Comment (yantosca) [reply run/GCClassic/createRunDir.sh]: Now fixed in commit c273374

Comment (yantosca) [reply run/GCClassic/createRunDir.sh]: Now fixed in commit 610d00b

File: CHANGELOG.md (hunk 1 above)
Comment (yantosca) [reply CHANGELOG.md]: Now fixed in commit c273374

**Comment by yantosca** [comment]
@msulprizio: I have merged these fixes into dev/14.2.1 locally and am running integration tests.  Results to follow.

**Comment by yantosca** [comment]
All GCHP integration tests passed
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #d5180cc GEOS-Chem submod update: Merge PR #1957 and PR #1958
GEOS-Chem #5d6b1f53e Merge PR #1963 (Add fixes for single-species carbon simulation )
HEMCO     #daf54d0 Merge PR #235 (Bug fix for vertical regridding)

Number of execution tests: 5

Submitted as SLURM job: 2568462
==============================================================================
 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
And all were zero-diff w/r/t the prior integration test.

**Review by msulprizio — APPROVED** [review]
Looks good to merge now!
